### PR TITLE
Dev Card: Add new actions for mocking user roles & permissions 

### PR DESCRIPTION
**Pull request by dereksmart**
This PR adds some more options to the Dev Card, in order to change/mock the state to reflect certain user permissions. 

screen shot 2017-02-27 at 11 13 45 am

New action type `MOCK_SWITCH_USER_PERMISSIONS` that passes the data for all `currentUser` data.  These actions update the state for both `initialState`, and `connection` branches, as some of our selectors are pulling from one, and some the other.  It's uncovering some messiness that we should look at in the future.  

I've also removed the `componentWillMount` function from the `<NonAdminView` which forced an update to the initial state, which was preventing the `editor` role to be able to mock.  I see no reason for setting that state on mount anyway.  Please correct me if I'm wrong.  

Some of the options are conditional.  For example, all admin levels will always be able to view stats, per the stats module options.  Master users will not be able to unlink, so that option is removed as well.

### CODE REVIEW

**Comment by eliorivero** [comment]
Here's a flow that fails:

1. start in Dashboard as any admin
2. go to Settings
3. switch to Editor

**Comment by dereksmart** [comment]
@eliorivero that reflects a bug in the current state of the Settings, since there is nothing that is currently handling the non-admin views in `feature/settings-overhaul` :) 

Related issue #6475

File: _inc/client/components/dev-card/index.jsx [hunk 1]
```diff
@@ -8,19 +8,110 @@ import classNames from 'classnames';
...
+import { isCurrentUserLinked } from 'state/connection';
+import {
+	switchPlanPreview,
+	canDisplayDevCard,
+	disableDevCard,
+	switchUserPermission
+} from 'state/dev-version';
 import Card from 'components/card';
 
 export const DevCard = React.createClass( {
```
Comment (eliorivero) [inline _inc/client/components/dev-card/index.jsx]: I didn't notice this when you originally built this. Maybe we should make a vow, a promise or an oath to never write `React.createClass` again and instead create the components either as functions, if they're stateless, or as ES6 classes. This is so ESLint doesn't continue adding warnings to the list.

**Comment by eliorivero** [comment]
Let's merge this. Will be nice to test #6475

File: _inc/client/components/dev-card/index.jsx [hunk 2]
```diff
@@ -8,19 +8,110 @@ import classNames from 'classnames';
...
+	maybeShowStatsToggle() {
+		if ( ! this.props.isAdmin ) {
+			return (
+				<div>
+					<hr/>
+					<ul>
+						<li>
+							<label>
+								<input
+									type='radio'
```
Comment (eliorivero) [inline _inc/client/components/dev-card/index.jsx]: Can you quickly update these `'` to be `"`? It introduced 16 new warnings to ESLint.

**Comment by dereksmart** [comment]
Fixed the eslint warnings... merging!

**Pull request merged**